plutus-contract/src/Wallet/Emulator/MultiAgent.hs

```diff
@@ -328,3 +342,6 @@ isValidated txn = do
     if notElem txn (join $ emState ^. chainState . Chain.chainNewestFirst)
         then throwError $ GenericAssertion $ "Txn not validated: " <> T.pack (show txn)
         else pure ()
+
+_singleton :: Prism' [a] a
```

Comment:
Possibly we just need `Review` and this could be [unto](http://hackage.haskell.org/package/lens-4.18.1/docs/Control-Lens-Review.html#v:unto) applied to singleton? Just guessing here from trawling `lens`. Not a big deal, though.

Reply:
Yeah I'll try to improve the prism situation. 

`unto = un . to`: of course!

Reply:
You were right, we just need a `Review` in many places where we currently require a `Prism`. I changed it accordingly. But maybe `_singleton` should stay a prism, since it is stronger than just a `Review`?